File: test/lang/1-pascal/fib_2.pas
library fibs;

Uses sysutils;


const
	version = 2;


function fibalgo (value: Int64) : Int64;
var
  l,p,n,i: Int64;
begin
  l := 0;
  p := 1;
  n := value;
  for i := 2 to value do
  begin
    n := l + p;
    l := p;
    p := n;
  end;
  fibalgo := n
end;


function fib (value: Int64) : Int64;
  cdecl; export;
begin
  fib := fibalgo(value)
end;

exports
  fib;


procedure printfib (value: Int64);
   cdecl; export;
var
  r: Int64;
  fmt,s : string;
begin
  r:=fibalgo(value);
  fmt:='[Pascal Fibonacci Library v%d] fib(%d) = %d';
  s:=Format(fmt,[version,value,r]);
  Writeln(s);
  Flush(Output);
end;

exports
  printfib;

end.

File: example/fib_2.c
#include "fib.h"

const unsigned short version = 2;

unsigned long fib(unsigned long value) {
	unsigned long f[value + 2];
	for (unsigned long n = 0; n <= value; n++)
		f[n] = n > 1 ? f[n - 1] + f[n - 2] : n;
	return f[value];
}

int print_library_info(FILE *stream) {
	return fprintf(stream, "[using Fibonacci library v%u: O(n))]\n", version);
}
